guis/desktop/gui.py

```diff
@@ -267,10 +313,14 @@ def draw_crs_bbox(self, crs):
     # Utilities
     def query(self):
         # Load all features drawn
-        features = self.json["features"]
+        # Handle error when switching logical operators and no geometry is drawn
+        try:
```

Comment:
This case is not really an exception. It's our design. Handling exceptions are usually expensive, so we want to avoid them if possible.